server/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -268,8 +272,59 @@ private void addParagraphRange(ParserRuleContext ctx) {
   @Override
   public List<Node> visitFileDescriptionEntry(FileDescriptionEntryContext ctx) {
     areaAWarning(ctx.getStart());
-    outlineTreeBuilder.addNode(ctx.fileName().getText(), NodeType.FILE, ctx);
-    return visitChildren(ctx);
+    String fileControlClause = "";
+    long fileControlMatchCount =
+        fileControls.keySet().stream().filter(ctx.cobolWord().getText().toUpperCase()::equals).count();
+    if (fileControlMatchCount != 1) {
+      addNoFileControlError(ctx);
+    } else {
+      FileControlEntryContext fileControlEntryContext = fileControls.remove(ctx.cobolWord().getText().toUpperCase());
+      Interval interval = new Interval(
+              fileControlEntryContext.fileControlClause(0).start.getStartIndex(),
+              fileControlEntryContext.fileControlClause(fileControlEntryContext.fileControlClause().size() - 1).stop.getStopIndex());
```

Comment:
Why not the range of `fileControlEntry`?

Reply:
Made changes to grammar as suggested